src/platform/update/new_release_monitor.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "new_release_monitor.h"
+
+#include "version.h"
+#include <multipass/exceptions/download_exception.h>
+#include <multipass/logging/log.h>
+#include <multipass/url_downloader.h>
+
+#include <fmt/format.h>
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QThread>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+
+namespace
+{
+constexpr auto update_url = "https://api.github.com/repos/CanonicalLtd/multipass/releases/latest";
+constexpr auto timeout = std::chrono::minutes(1);
+
+QJsonObject parse_manifest(const QByteArray& json)
+{
+    QJsonParseError parse_error;
+    const auto doc = QJsonDocument::fromJson(json, &parse_error);
+    if (doc.isNull())
+        throw std::runtime_error(parse_error.errorString().toStdString());
+
+    if (!doc.isObject())
+        throw std::runtime_error("invalid JSON object");
+    return doc.object();
+}
+} // namespace
+
+class mp::LatestReleaseChecker : public QThread
+{
+    Q_OBJECT
+public:
+    void run()
+    {
+        QUrl url(update_url);
+        try
+        {
+            mp::URLDownloader downloader(timeout);
+            QByteArray json = downloader.download(url);
+
+            const auto manifest = ::parse_manifest(json);
+            mp::NewReleaseInfo release;
+            release.version = manifest["tag_name"].toString();
+            release.url = manifest["html_url"].toString();
+
+            mpl::log(mpl::Level::debug, "update",
+                     fmt::format("Latest Multipass release available is version {}", qPrintable(release.version)));
+
+            emit latest_release_found(release);
+        }
+        catch (const mp::DownloadException& e)
+        {
+            mpl::log(mpl::Level::info, "update", fmt::format("Failed to check for updates: {}", qPrintable(e.what())));
```

Comment:
I don't have strong opinion on this, but I thought warning was ok as an update check failure doesn't compromise the user experience, and most typically it'll print if user offline. 

This is my head-classification of log levels: warnings are for "this might cause you problems", error: "this is broken", info: "you won't experience a problem, but you might like to know this" and debug "useful info for developers"

Reply:
@Saviq can you tie-break us here please?

Reply:
I think `info` is enough here.